src/main/java/org/jetbrains/research/groups/ml_methods/refactoring/detection/RMiner.java

```diff
@@ -0,0 +1,101 @@
+        List<Integer> refactoringsNumbersInCommit = new ArrayList<>();
+        miner.detectAll(repository, branch, new RefactoringHandler() {
+            private int processedCommitsNumber = 0;
+
+            @Override
+            public void handle(RevCommit commitData, List<Refactoring> refactorings) {
+                LOGGER.info("Processing new commit. Commit hash: " + commitData.getId().getName());
+                if (refactorings.size() != 0) {
+                    refactoringsNumbersInCommit.add(refactorings.size());
+                }
+                for (Refactoring refactoring : refactorings) {
+                    LOGGER.info("Refactoring: " + refactoring.toString());
+                    MoveOperationRefactoring moveOperationRefactoring = (MoveOperationRefactoring) refactoring;
+                    String targetClassQualifiedName = moveOperationRefactoring.getMovedOperation().getClassName();
+                    String originalClassQualifiedName = moveOperationRefactoring.getOriginalOperation().getClassName();
+                    String movedMethodName = moveOperationRefactoring.getMovedOperation().getName();
+                    String originalMethodName = moveOperationRefactoring.getOriginalOperation().getName();
+                    List<String> movedParamsClassesQualifiedNames =
+                            moveOperationRefactoring.getMovedOperation().getParametersWithoutReturnType().stream()
+                                    .map(umlParameter -> umlParameter.getType().toString())
+                                    .collect(Collectors.toList());
+                    List<String> originalParamsClassesQualifiedNames =
+                            moveOperationRefactoring.getOriginalOperation().getParametersWithoutReturnType().stream()
+                                    .map(umlParameter -> umlParameter.getType().toString())
+                                    .collect(Collectors.toList());
+                    detectedRefactorings.add(
+                            new MoveMethodRefactoringFromVCS(
+                                    repositoryUrl,
+                                    commitData.getId().getName(),
+                                    targetClassQualifiedName,
+                                    originalClassQualifiedName,
+                                    originalMethodName,
+                                    movedMethodName,
+                                    originalParamsClassesQualifiedNames,
+                                    movedParamsClassesQualifiedNames
+                            )
+                    );
+                }
+                processedCommitsNumber++;
+                System.out.println("Processed commits: " + processedCommitsNumber + " / " + commitsNumber);
+            }
+        });
+        RefactoringDetectionExecutionInfo executionInfo =
+                new RefactoringDetectionExecutionInfo(
```

Comment:
Seems like it's responsibility of `RepositoryDetectedRefactorings` to create `RefactoringDetectionExecutionInfo` inside itself. Otherwise it might lead to misuse.

Reply:
It is a bad practice to pass a lot of parameters. I don't know to what misuse it might lead to. If you have execution info then you collect it to object and this object can calculate some statistics for you.